### ISSUE TITLE
`eslint --init` fails

### ISSUE THREAD

**Issue by KINFOO** [eslint/eslint#7211]
* **ESLint Version:** v3.5.0
* **Node Version:** v4.5.0
* **npm Version:** 2.15.9

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`--init` failed, I have no configuration.

**What did you do? Please include the actual source code causing the issue.**
Well, I typed `$ eslint --init` in console.

**What did you expect to happen?**
Create a setup a configuration file.

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
$ eslint --init 
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? path
? What format do you want your config file to be in? JavaScript
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser, Node
? Do you use CommonJS? No
? Do you use JSX? No
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s 

Automatic Configuration failed.  No files were able to be parsed.
Error: Automatic Configuration failed.  No files were able to be parsed.
    at configureRules (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:142:15)
    at processAnswers (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:254:18)
    at null.completed (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:414:30)
    at PromptUI.onCompletion (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/lib/ui/prompt.js:57:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1550:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1489:14)
    at Subject.Rx.Subject.addProperties.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:5871:19)
    at Subject.tryCatcher (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:63:31)
    at AutoDetachObserverPrototype.completed (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:5796:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1489:14)
```
I am available for further details.
Thanks for the good work.

**Comment by platinumazure** [comment]
Do you have any JavaScript files already created in the folder (or any subfolders thereof) where you were running the `eslint --init` command? The "Inspect your JavaScript files" method of creating config requires that at least one parseable file be present in the folder or in subfolders.

**Comment by eslintbot** [comment]
Hi @KINFOO, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by KINFOO** [comment]
@platinumazure Hi,

I was not the case.

I just ran `eslint --init` in a folder which goes like:

```sh
$ ls
app.js  bower.json  build  config  Gruntfile.js  lib  logs  package.json  public  README.md  routes  run.sh  views
```

At least `Gruntfile.js` is parseable, but ther is more in `public/`.

```sh
$ node Gruntfile.js # Returns silently, so parseable, right?
$ find public/ -name "*.js" -type f | wc --lines 
1096
```

Well same __error__.

```sh
$ eslint --init 
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? path
? What format do you want your config file to be in? JavaScript
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser, Node
? Do you use CommonJS? No
? Do you use JSX? No
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s 

Automatic Configuration failed.  No files were able to be parsed.
Error: Automatic Configuration failed.  No files were able to be parsed.
    at configureRules (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:142:15)
    at processAnswers (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:254:18)
    at null.completed (/home/kevin/.local/lib/node_modules/eslint/lib/config/config-initializer.js:414:30)
    at PromptUI.onCompletion (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/lib/ui/prompt.js:57:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1550:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1489:14)
    at Subject.Rx.Subject.addProperties.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:5871:19)
    at Subject.tryCatcher (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:63:31)
    at AutoDetachObserverPrototype.completed (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:5796:56)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/home/kevin/.local/lib/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1489:14)
```

Thanks for your time

**Comment by platinumazure** [comment]
Ah, please forgive me, I misread your initial post.

```
? Which file(s), path(s), or glob(s) should be examined? path
```

I think you might be misunderstanding this question. This question isn't asking whether you prefer to examine files vs paths vs globs, but rather you need to enter all appropriate files, paths, or globs for ESLint to examine. My guess is "path" is being interpreted as a filename or folder to such, and when ESLint can't find either, it fails.

What you probably want is either `.` (current folder), `**/*.js` (glob pattern that matches all JavaScript files in current folder and subfolders), or `public` (if you don't want the Gruntfile or app.js to influence the generated ESLint config).

Give that a try and let me know if that works better for you.

**Comment by nzakas** [comment]
Yes, @platinumazure is correct. You need to provide a path like `*.js` or `app.js` for `--init` to work.

**Comment by KINFOO** [comment]
```sh
$ eslint --init                                
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? /w/av-ui
? What format do you want your config file to be in? JavaScript
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser, Node
? Do you use CommonJS? No
? Do you use JSX? No
Determining Config: 100% [==============================] 79.0s elapsed, eta 0.0s 


Enabled 124 out of 226 rules based on 1179 files.
Successfully created .eslintrc.js file in /home/kevin
```

![Thank You](https://i.imgflip.com/1b4i5f.jpg)

**Comment by 462960** [comment]
I have the same problem. What's the meaning of  /w/av-ui?

**Comment by KINFOO** [comment]
@462960 An actual folder with JavaScript files.
